lib/net/ssh/authentication/methods/abstract.rb

```diff
@@ -21,12 +21,16 @@ class Abstract
           # this.
           attr_reader :key_manager
 
+          # Disable rsa SHA-2 auth, usefull for tests
+          attr_reader :rsa_sha2_auth_disable
```

Comment:
I'm not agree to had this, if it's only for units test. The best way is to disable in the test suite on server side. By the ways, you need to add more test on the PR.